lib/fog/libvirt/models/compute/templates/server.xml.erb

```diff
@@ -32,13 +32,38 @@
 <% end -%>
   <clock offset='utc'/>
   <devices>
+<% args = {}
+   if File.file?('/etc/foreman/ceph.conf')
+     File.readlines('/etc/foreman/ceph.conf').each do |line|
+       pair = line.strip.split("=")
+       key = pair[0]
+       value = pair[1]
+       args[key] = value
+     end
+   end
+%>
 <% volumes.each do |vol| -%>
+  <% if File.file?('/etc/foreman/ceph.conf') && vol.pool_name.include?(args["libvirt_ceph_pool"]) %>
+    <disk type='network' device='disk'>
+      <driver name='qemu' type='<%= vol.format_type %>' cache='writeback' discard='unmap'/>
+      <source protocol='rbd' name='<%= vol.path %>'>
+        <% args["monitor"].split(",").each do |mon| %>
+        <host name='<%= mon %>' port='<%= args["port"] %>'/>
+        <% end %>
+      </source>
+      <auth username='<%= args["auth_username"] %>'>
+        <secret type='ceph' uuid='<%= args["auth_uuid"] %>'/>
+      </auth>
+      <target dev='sd<%= ('a'..'z').to_a[volumes.index(vol)] %>' bus='scsi'/>
```

Comment:
Why does this use `scsi` instead of `virtio`? `virtio` appears to work fine for me and the regular disk block below also uses it.

Reply:
`virtio` does not support discard / fstrim yet (unless you use very recent kernels). `scsi` however is mature, similar in performance, has device names matching what is encountered on a bare-metal host, and supports discard on LTS distributions, which grants orders of magnitude of space savings if used with Ceph RBD.